scripts/zones/Xarcabard/Zone.lua

```diff
@@ -43,7 +43,15 @@ zoneObject.onZoneIn = function(player, prevZone)
     elseif quests.rainbow.onZoneIn(player) then
         cs = 9
     elseif unbridledPassionCS == 3 then
-        cs = 4
+        if
+            math.floor(pos.x) == math.floor(xi.teleport.destination[xi.teleport.id.VAHZL][1]) and
```

Comment:
I think a better way to do this would just be a distance check with the Vahzl teleport. With double/float numbers, checking against zero is one thing, but this flooring/checking pattern doesn't sit right.

Reply:
yeah a friend suggested that too. It's fixed now =)